### PR TITLE
More Precise Client Hellos

### PR DESCRIPTION
**Pull request by bvandersloot**
Use new ztls configuration options to create more exactly matching client hello messages.

### CODE REVIEW

**Comment by sergeyfrolov** [comment]
Thank you for this important contribution!
Did you check that it actually matches Android4.4, Firefox50 and Chrome47*(with that exception) on the wire?

**Comment by bvandersloot** [comment]
I had to make one small fix to get the Android fingerprint to send. Unfortunately, Android will negotiate a non-gcm cipher suite with some webservers. This is not in the compatible list. This limits the set of servers Android formatted client-hellos can be sent to.

But the ClientHellos look to be the same on the wire.

File: tapdance/mimic_client_hello.go [hunk 1]
```diff
@@ -48,20 +51,204 @@ var TDFirefox50Ciphers = []uint16{
...
 		fallthrough
 	case "Firefox50":
-		return ztls.Config{
-			ForceSessionTicketExt:  true,
-			CipherSuites:           TDFirefox50Ciphers,
-			SessionTicketsDisabled: false,
-			NextProtos:             []string{"h2", "http/1.1"},
-			ExtendedMasterSecret:   true,
+		conf := ztls.Config{
+			InsecureSkipVerify: true,
```
Comment (sergeyfrolov) [inline tapdance/mimic_client_hello.go:67]: Why?

**Review by sergeyfrolov — COMMENTED** [review]
Yeah, we should not use fingerprints that offer CHACHA20_POLY1305 and CBC ciphers before GCM.

**Comment by sergeyfrolov** [comment]
Maybe we should use different Android fingerprint, that would prefer GCM?

**Comment by bvandersloot** [comment]
I will dig around in different Android fingerprints for ones that agree more with GCM.

File: tapdance/mimic_client_hello.go (hunk 1 above)
Comment (sergeyfrolov) [reply tapdance/mimic_client_hello.go:67]: Actually, given #18 this is fine, but we should totally remove this flag asap

**Comment by sergeyfrolov** [comment]
Nice, thank you!
Meanwhile, I can merge this.
Firefox and Chrome versions looks a bit outdated now, but we can work on it iteratively as well.

**Pull request merged**